cmd/id-exporter/main_test.go

```diff
@@ -61,24 +61,102 @@ func TestFindIDs(t *testing.T) {
 	// *not* be present since their certificate has already expired. Unlike
 	// previous versions of this test RegD is not filtered out for having a `tel:`
 	// contact field anymore - this is the duty of the notify-mailer.
-	ids, err = testCtx.c.findIDs()
+	results, err = testCtx.c.findIDs()
 	test.AssertNotError(t, err, "findIDs() produced error")
-	test.AssertEquals(t, len(ids), 3)
-	test.AssertEquals(t, ids[0].ID, regA.ID)
-	test.AssertEquals(t, ids[1].ID, regC.ID)
-	test.AssertEquals(t, ids[2].ID, regD.ID)
+	test.AssertEquals(t, len(results), 3)
+	for _, entry := range results {
+		switch entry.ID {
+		case 1:
+			test.AssertEquals(t, entry.ID, regA.ID)
```

Comment:
case could just be: `case regA.ID:` and so forth for 2 and 4. Then add a default that does `test.Error("Unexpected ID: %d", entry.ID)`.

This applies to all these constructions. Basically there's nothing to assert beyond that entry.ID is in the case statements, except for the one that also asserts the hostname found, so keep those AssertEquals, and otherwise just permit the IDs that match your expectations. It'll be easier to read, to boot!

Reply:
Yeah this makes sense. I've gone ahead and made the changes.